### PR TITLE
[@types/node] Update `v8` type definitions

### PR DESCRIPTION
**Pull request by ghost**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test node`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
  - https://nodejs.org/docs/latest-v20.x/api/v8.html#startup-snapshot-api
  - https://nodejs.org/docs/latest-v20.x/api/v8.html#v8setheapsnapshotnearheaplimitlimit
  - https://nodejs.org/docs/latest-v20.x/api/v8.html#v8writeheapsnapshotfilenameoptions
  - https://nodejs.org/docs/latest-v20.x/api/v8.html#v8getheapsnapshotoptions
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Review by Semigradsky — APPROVED** [review]

File: types/node/v16/ts4.8/v8.d.ts [hunk 1]
```diff
@@ -455,17 +498,125 @@ declare module "v8" {
...
+        /**
+         * Returns true if the Node.js instance is run to build a snapshot.
+         * @since v16.17.0
+         */
+        isBuildingSnapshot(): boolean;
+    }
+    /**
+     * The `v8.startupSnapshot` interface can be used to add serialization and deserialization hooks for custom startup snapshots.
+     *
+     * ```sh
```
Comment (codershiba) [inline types/node/v16/ts4.8/v8.d.ts:541]:
```suggestion
     * ```bash
```

File: types/node/v8.d.ts [hunk 2]
```diff
@@ -629,6 +648,114 @@ declare module "v8" {
...
+     *   const lang = process.env.BOOK_LANG || 'en_US';
+     *   const book = process.argv[1];
+     *   const name = `${book}.${lang}.txt`;
+     *   console.log(shelf.storage.get(name));
+     * }, shelf);
+     * ```
+     *
+     * The resulted binary will get print the data deserialized from the snapshot during start up, using the refreshed `process.env` and `process.argv` of the launched process:
+     *
+     * ```sh
```
Comment (codershiba) [inline types/node/v8.d.ts:748]:
```suggestion
     * ```bash
```

File: types/node/v8.d.ts [hunk 3]
```diff
@@ -629,6 +648,114 @@ declare module "v8" {
...
+        /**
+         * Returns true if the Node.js instance is run to build a snapshot.
+         * @since v18.6.0, v16.17.0
+         */
+        isBuildingSnapshot(): boolean;
+    }
+    /**
+     * The `v8.startupSnapshot` interface can be used to add serialization and deserialization hooks for custom startup snapshots.
+     *
+     * ```sh
```
Comment (codershiba) [inline types/node/v8.d.ts:680]:
```suggestion
     * ```bash
```

**Review by peterblazejewicz — APPROVED** [review]
@codershiba thanks!

File: types/node/v16/ts4.8/v8.d.ts [hunk 4]
```diff
@@ -455,17 +498,128 @@ declare module "v8" {
...
```
Comment (codershiba) [inline types/node/v16/ts4.8/v8.d.ts:543]:
```suggestion
     * ```bash
```

File: types/node/v8.d.ts [hunk 5]
```diff
@@ -629,6 +648,116 @@ declare module "v8" {
...
+     *   const lang = process.env.BOOK_LANG || 'en_US';
+     *   const book = process.argv[1];
+     *   const name = `${book}.${lang}.txt`;
+     *   console.log(shelf.storage.get(name));
+     * }, shelf);
+     * ```
+     *
+     * The resulted binary will get print the data deserialized from the snapshot during start up, using the refreshed `process.env` and `process.argv` of the launched process:
+     *
+     * ```sh
```
Comment (codershiba) [inline types/node/v8.d.ts:750]:
```suggestion
     * ```bash
```

File: types/node/v8.d.ts [hunk 6]
```diff
@@ -629,6 +648,116 @@ declare module "v8" {
...
+        /**
+         * Returns true if the Node.js instance is run to build a snapshot.
+         * @since v18.6.0, v16.17.0
+         */
+        isBuildingSnapshot(): boolean;
+    }
+    /**
+     * The `v8.startupSnapshot` interface can be used to add serialization and deserialization hooks for custom startup snapshots.
+     *
+     * ```sh
```
Comment (codershiba) [inline types/node/v8.d.ts:682]:
```suggestion
     * ```bash
```

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/node/ts4.8/v8.d.ts
1. types/node/v8.d.ts
1. types/node/v16/v8.d.ts
1. types/node/v18/v8.d.ts
1. types/node/v16/ts4.8/v8.d.ts

<details>
<summary>as well as these 1 other files...</summary>
<p>types/node/v18/ts4.8/v8.d.ts</p>
</details>


Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 15291d3591f107a96be91d91a50db4b8ae686ee3
</p>

**Review by peterblazejewicz — APPROVED** [review]

**Comment by codershiba** [comment]
Ready to merge

**Pull request merged**